src/client/client.cpp

```diff
@@ -309,10 +309,10 @@ void Client::Stop()
 	if (m_localdb) {
 		infostream << "Local map saving ended." << std::endl;
 		m_localdb->endSave();
+		m_localdb.reset();
 	}
 
-	if (m_mods_loaded)
```

Comment:
Any reason to only delete it when `m_mods_loaded == true` ? because it means the rest of the time this is a potential leak.

Reply:
`m_script` should not exist when no mods are loaded, but you can remove the condition.

Reply:
It _should not_ exist, but by watching code, there is an execution path were it could. `Client::loadMods()` has a return path line 214 `if (!modconf.isConsistent())`  which would leave existing m_script without loaded mods.

Reply:
I assume this is an oversight. I'd just go ahead with the alternative AND setting `m_script = nullptr;`.